### PR TITLE
Star experiments in table

### PR DESCRIPTION
**Pull request by wolmir**
This is the first PR for #1676 

This PR will introduce the Star Experiments feature to the experiments table webview.

This option will be available in the form of star icons in the first cells, as well as context-menu options.

Demo:

https://user-images.githubusercontent.com/1231848/176021943-5f744280-3eb3-4f3c-833a-3bee0948d84e.mov

### CODE REVIEW

**Review by rogermparent — CHANGES_REQUESTED** [review]
There's an issue with the context menu where invoking it on a multi-select that contains both starred and unstarred items will create an empty context menu:

https://user-images.githubusercontent.com/9111807/176047323-f3b19880-a1b4-45c1-88a4-96e3bc59d12d.mp4

Beyond that, I think we'll want to show star/unstar options in this situation, especially if we add more multi-select options like selecting ranges or groups. 
I think the most useful logic would be simply showing group star if any are unstarred and group unstar if any are starred, allowing for both to be present. Having a single option would make it very inconvenient for users to use the option that has less priority, especially where the selection clears when any option is taken.

File: extension/src/experiments/model/index.ts
Comment (mattseddon) [inline extension/src/experiments/model/index.ts:124]: [Q] Do you think there will be some in-between state in the future? Why not start with a boolean value?

File: extension/src/experiments/model/index.ts
Comment (mattseddon) [inline extension/src/experiments/model/index.ts:551]: [C] Changing to a boolean could simplify this method.

File: extension/src/experiments/model/index.ts
Comment (mattseddon) [inline extension/src/experiments/model/index.ts:571]: [Q] Is `addSelected` used anymore? Can we combine these methods?

Comment (mattseddon) [reply extension/src/experiments/model/index.ts:571]: rename to `addDetails`?

File: webview/src/experiments/components/table/Row.tsx
Comment (mattseddon) [inline webview/src/experiments/components/table/Row.tsx:137]: [Q] Should there be dividers between the VARY_AND_ commands and toggle star and also remove?

File: webview/src/experiments/components/App.test.tsx
Comment (mattseddon) [inline webview/src/experiments/components/App.test.tsx:756]: [Q] New test for new behavior?

File: extension/src/experiments/model/index.ts
Comment (mattseddon) [inline extension/src/experiments/model/index.ts:86]: [Q] What will happen if we move between branches/commits and experiments are still starred? I can't see anything that would ever flush stale experiments out of this list. An ever-expanding object is something that we should try to avoid.

Comment (shcheklein) [reply extension/src/experiments/model/index.ts:87]: I don't see a better option tbh, any ideas? I hope this list won't be growing too fast?

It should be probably split by commits for faster access though.

**Comment by wolmir** [comment]
@rogermparent @maxagin 

What about something like this for heterogeneous rows?

https://user-images.githubusercontent.com/1231848/176250520-3d656e54-69d0-481a-b8cd-14c816100807.mov

**Comment by wolmir** [comment]
@rogermparent @mattseddon Can you re-review this please?

**Review by rogermparent — APPROVED** [review]
The issue I found was handled nicely, and from a user standpoint feature works enough that I'd consider it good enough to approve. I'm not sure about the other concerns, those may have been handled in the meeting but otherwise we may want to wait for Matt's approval before merging.

File: extension/src/experiments/model/index.ts
Comment (mattseddon) [reply extension/src/experiments/model/index.ts:566]: [N] Could be 

```
return { ...this.addSelected(experiment), starred: !!this.isStarred(id) }
```

**Review by mattseddon — APPROVED** [review]

**Comment by mattseddon** [comment]
@wolmir after you merge this tomorrow please make a release to the marketplace 👍🏻.

**Pull request merged**
